app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -107,6 +143,74 @@ fun WallpaperSettings(
     }
 }
 
+/**
+ * A group of [Wallpaper]s.
+ *
+ * @property wallpapers The list of [Wallpaper]s in this group.
+ * @property title An optional title for this group.
+ * @property limit The maximum number of wallpapers that will be shown in this group.
+ * @property id A unique ID for this group (uses a generated UUID by default)
+ */
+
+data class WallpaperProviderGroup(
+    var wallpapers: List<Wallpaper>,
+    var title: String? = null,
+    val limit: Int = Integer.MAX_VALUE,
+    val id: String = UUID.randomUUID().toString(),
+)
```

Comment:
I don't think this data class will be necessary if we decide to use an extension function to group things as I suggested above.